### PR TITLE
Experience/9403/facilities providers

### LINKED ISSUE #9403: STLT facilities/providers page - Hook up API
## Problem statement

As part of the STLT Dashboard, we need to implement the API of the facilities/providers index page. 

## What you need to know

 

- [ ] hook up to the API (PR ???)

## To do

- [ ] Add/Update interface
- [ ] Add/Update hook
- [ ] Add endpoint call
- [ ] Unit tests

### PR DESCRIPTION
**Pull request by penny-lischer**
This PR ...

Hooks up the API to Data Dashboard UI

Test Steps:
1. Log into RS
2. Select an organization at the State level
3. Change URL to <server>/data-dashboard
4. Click on the All facilities & providers link
5. View changes

## Changes
- Added API hook
- Added table pagination and sorting
- Hooked up the filter
- Added unit tests

Screenshot 2023-07-24 at 9 50 38 AM

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [x] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

## Linked Issues
- Fixes #9403

### CODE REVIEW

**Review by jpandersen87 — APPROVED** [review]

**Review by KeithNava — APPROVED** [review]

File: frontend-react/src/components/DataDashboard/FacilitiesProviders/FacilitiesProviders.module.scss [hunk 1]
```diff
@@ -8,19 +8,23 @@
...
 
     hr {
-        border: 1px solid #73B3E7;
+        border: 1px solid #73b3e7;
     }
 
     :global {
         .usa-breadcrumb {
-            background-color: #D9E8F6;
+            background-color: #d9e8f6;
```
Comment (etanb) [inline frontend-react/src/components/DataDashboard/FacilitiesProviders/FacilitiesProviders.module.scss:16]: can we use the color token? `blue-10`

File: frontend-react/src/components/DataDashboard/FacilitiesProviders/FacilitiesProviders.module.scss [hunk 2]
```diff
@@ -8,19 +8,23 @@
     }
 
     hr {
-        border: 1px solid #73B3E7;
+        border: 1px solid #73b3e7;
     }
```
Comment (etanb) [inline frontend-react/src/components/DataDashboard/FacilitiesProviders/FacilitiesProviders.module.scss:12]: can we use the color token? `blue-30`

**Comment by jillian-hammer** [comment]
Design review notes, thank you @penny-lischer! 

- [ ] The "Data Dashboard" breadcrumb link color is `white` (`#FFFFFF`), should be `primary` (`#005EA2`) image
- [ ] The "All facilities and providers" text should be `bold` (H1 style in Figma)
- [ ] Increase `.padding-bottom-3` from `1.5rem` to `2.5rem`

File: frontend-react/src/components/DataDashboard/FacilitiesProviders/FacilitiesProviders.module.scss (hunk 1 above)
Comment (penny-lischer) [reply frontend-react/src/components/DataDashboard/FacilitiesProviders/FacilitiesProviders.module.scss:16]: I tried to and it doesn't get read from this file.

**Comment by penny-lischer** [comment]
> Design review notes, thank you @penny-lischer!
> 
> * [x]  The "Data Dashboard" breadcrumb link color is `white` (`#FFFFFF`), should be `primary` (`#005EA2`) image
> * [x]  The "All facilities and providers" header should be `bold` (H1 style in Figma)
> * [x]  Increase `.padding-bottom-3` from `1.5rem` to `2.5rem`

@jillian-hammer Please re-review in trialfrontend03

**Comment by jillian-hammer** [comment]
> > Design review notes, thank you @penny-lischer!
> > 
> > * [x]   The "Data Dashboard" breadcrumb link color is `white` (`#FFFFFF`), should be `primary` (`#005EA2`) image
> > * [x]   The "All facilities and providers" header should be `bold` (H1 style in Figma)
> > * [x]   Increase `.padding-bottom-3` from `1.5rem` to `2.5rem`
> 
> @jillian-hammer Please re-review in trialfrontend03

looks great! thank you @penny-lischer !

**Pull request merged**